pkg/controller/proxyconfig/controller.go

```diff
@@ -67,72 +90,145 @@ type ReconcileProxyConfig struct {
+		clusterCfgMap := &corev1.ConfigMap{}
+
+		log.Printf("Reconciling proxy '%s'", request.Name)
+		if err := r.client.Get(context.TODO(), request.NamespacedName, proxyConfig); err != nil {
+			if apierrors.IsNotFound(err) {
+				// Request object not found, could have been deleted after reconcile request.
+				// Return and don't requeue
+				log.Println("proxy not found; reconciliation will be skipped", "request", request)
+				return reconcile.Result{}, nil
+			}
+			// Error reading the object - requeue the request.
+			return reconcile.Result{}, fmt.Errorf("failed to get proxy '%s': %v", request.Name, err)
+		}
+
+		// A nil proxy is generated by upgrades and installs not requiring a proxy.
+		validate := true
+		if !isSpecHTTPProxySet(&proxyConfig.Spec) &&
+			!isSpecHTTPSProxySet(&proxyConfig.Spec) &&
+			!isSpecNoProxySet(&proxyConfig.Spec) {
+			log.Printf("httpProxy, httpsProxy and noProxy not defined for proxy '%s'; validation will be skipped",
+				request.Name)
+			validate = false
+		}
+
+		// Only proceed if the required config objects can be collected.
+		if validate {
+			if err := r.client.Get(context.TODO(), types.NamespacedName{Name: names.PROXY_CONFIG}, infraConfig); err != nil {
+				log.Printf("failed to get infrastructure config '%s': %v", names.PROXY_CONFIG, err)
+				r.status.SetDegraded(statusmanager.ProxyConfig, "InfraConfigError",
+					fmt.Sprintf("Error getting infrastructure config %s: %v.", names.PROXY_CONFIG, err))
+				return reconcile.Result{}, nil
+			}
+			if err := r.client.Get(context.TODO(), types.NamespacedName{Name: names.PROXY_CONFIG}, netConfig); err != nil {
+				log.Printf("failed to get network config '%s': %v", names.PROXY_CONFIG, err)
+				r.status.SetDegraded(statusmanager.ProxyConfig, "NetworkConfigError",
+					fmt.Sprintf("Error getting network config '%s': %v.", names.PROXY_CONFIG, err))
+				return reconcile.Result{}, nil
+			}
+			if err := r.client.Get(context.TODO(), types.NamespacedName{Name: "cluster-config-v1", Namespace: "kube-system"},
+				clusterCfgMap); err != nil {
+				log.Printf("failed to get configmap '%s/%s': %v", clusterCfgMap.Namespace, clusterCfgMap.Name, err)
+				r.status.SetDegraded(statusmanager.ProxyConfig, "ClusterConfigError",
+					fmt.Sprintf("Error getting cluster config configmap '%s/%s': %v.", clusterCfgMap.Namespace,
+						clusterCfgMap.Name, err))
+				return reconcile.Result{}, nil
+			}
+			if err := r.ValidateProxyConfig(&proxyConfig.Spec); err != nil {
+				log.Printf("Failed to validate proxy '%s': %v", proxyConfig.Name, err)
+				r.status.SetDegraded(statusmanager.ProxyConfig, "InvalidProxyConfig",
+					fmt.Sprintf("The configuration is invalid for proxy '%s' (%v). "+
+						"Use 'oc edit proxy.config.openshift.io %s' to fix.", proxyConfig.Name, err, proxyConfig.Name))
+				return reconcile.Result{}, nil
+			}
+		}
+
+		// Update proxy status.
+		if err := r.syncProxyStatus(proxyConfig, infraConfig, netConfig, clusterCfgMap); err != nil {
```

Comment:
if `validate == false`, then  `infraConfig`, `netConfig`, and `clusterCfgMap` will be `nil` - is that OK?

Reply:
@squeed thanks for pointing this out. I'm going to pull the calls out of the `verify` block since the values are only being used for spec>status sync.